### PR TITLE
Gaussian splatting

### PR DESCRIPTION
**Pull request by dylanebert**
Introduction post to gaussian splatting

### CODE REVIEW

File: gaussian-splatting.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+2. Sort the gaussians by depth.
+3. For each pixel, iterate over each gaussian front-to-back, blending them together.
+
+Additional optimizations are described in [the paper](https://huggingface.co/papers/2308.04079).
+
+It's also essential that the rasterizer is differentiable, so that it can be trained with stochastic gradient descent. However, this is only relevant for training - the trained gaussians can also be rendered with a non-differentiable approach.
+
+## The future of graphics
+
+So what does this mean for the future of graphics? Well, let's break up it up into pros/cons:
```
Comment (pcuenca) [inline gaussian-splatting.md:104]:
```suggestion
So what does this mean for the future of graphics? Well, let's break it up into pros/cons:
```

File: gaussian-splatting.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+**Cons**
+1. High VRAM usage (4GB to view, 12GB to train)
+2. Large disk size (1GB+ for a scene)
+3. Incompatible with existing rendering pipelines
+3. Static (for now)
+
+So far, the original CUDA implementation has not been adapted to production rendering pipelines, like Vulkan, DirectX, WebGPU, etc, so it's yet to be seen what the impact will be.
+
+There have already been the following adaptions:
```
Comment (pcuenca) [inline gaussian-splatting.md:119]:
```suggestion
There have already been the following adaptations:
```

File: gaussian-splatting.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+1. [Remote viewer](https://huggingface.co/spaces/dylanebert/gaussian-viewer)
+2. [WebGPU viewer](https://github.com/cvlab-epfl/gaussian-splatting-web)
+3. [WebGL viewer](https://huggingface.co/spaces/cakewalk/splat)
+4. [Unity viewer](https://github.com/aras-p/UnityGaussianSplatting)
+
+These rely either on remote streaming (1) or a traditional quad-based rasterization approach (2-4).
+
+So will we see 3D Gaussian Splatting fully reimplemented in a production environment? The answer is *probably yes*. The primary bottleneck is sorting millions of gaussians, which is done efficiently in the original implementation using [CUB device radix sort](https://nvlabs.github.io/cub/structcub_1_1_device_radix_sort.html), a highly optimized sort only available in CUDA. However, with enough effort, it's certainly possible to achieve this level of performance in other rendering pipelines.
+
+If you have any questions or would like to get involved, join the [Hugging Face Discord](https://hf.co/join/discord)!
```
Comment (pcuenca) [inline gaussian-splatting.md:129]: Can we deep link to a channel, or mention it?

File: gaussian-splatting.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+So far, the original CUDA implementation has not been adapted to production rendering pipelines, like Vulkan, DirectX, WebGPU, etc, so it's yet to be seen what the impact will be.
+
+There have already been the following adaptions:
+1. [Remote viewer](https://huggingface.co/spaces/dylanebert/gaussian-viewer)
+2. [WebGPU viewer](https://github.com/cvlab-epfl/gaussian-splatting-web)
+3. [WebGL viewer](https://huggingface.co/spaces/cakewalk/splat)
+4. [Unity viewer](https://github.com/aras-p/UnityGaussianSplatting)
+
+These rely either on remote streaming (1) or a traditional quad-based rasterization approach (2-4).
```
Comment (pcuenca) [inline gaussian-splatting.md:125]: Maybe clarify that quad rasterization avoids the critical path of sorting the gaussians, but quality is worse.

**Review by pcuenca — COMMENTED** [review]
Very cool!

I'd maybe suggest to provide a bit more context on why this might be a big deal and/or how it compares to other techniques such as NeRF. Why is everyone excited, and why would you want to use it in a 3D pipeline (if it was possible) rather than using polygons?

My understanding is that each scene must be trained. How would this impact computer graphics systems such as games?

An idea for visualization could be to animate an increasing number of gaussians that render the same scene. Not sure if it's possible, and it doesn't have to be in the blog, we could communicate it later if we have it :)

File: gaussian-splatting.md (hunk 3 above)
Comment (dylanebert) [reply gaussian-splatting.md:129]: Currently we don't have a 3D channel, maybe would be good to make one?

**Pull request merged**